### ISSUE TITLE
Ruby syntax error missing reported as a zero-length range at EOF

### ISSUE THREAD

**Issue by korasino** [rubocop/rubocop#14980]
This happens when using Solargraph + RuboCop diagnostics: `Lint/Syntax` for a missing `end` is reported as a zero-length range at EOF (`start == end`), which makes inline highlighting unreliable in some editors (like [here](https://github.com/zed-extensions/ruby/issues/259)).

--------

## Steps to reproduce the problem

1. Create `test.rb` with:
```
if true
  puts 'x'
```
2. Use Solargraph (with RuboCop diagnostics enabled) and inspect `textDocument/publishDiagnostics` for this file.

Diagnostic payload received:
```
[
  {
    "range": {
      "start": { "line": 2, "character": 0 },
      "end": { "line": 2, "character": 0 }
    },
    "severity": 1,
    "source": "rubocop",
    "code": "Lint/Syntax",
    "message": " unexpected token $end\n(Using Ruby 2.7 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)"
  }
]
```
## Expected behavior

For `missing end`, the diagnostic range should be non-empty (or otherwise point to a visible span), so editors can reliably show inline error highlighting.

## Actual behavior

In Solargraph + RuboCop diagnostics, `Lint/Syntax` for `unexpected token $end` is reported as a zero-length range at EOF (`start == end`).

## RuboCop version

 1.85.0 (using Parser 3.3.10.2, rubocop-ast 1.49.0, analyzing as Ruby 2.7, running on ruby 3.4.8) [x86_64-linux]
